doc/Clusters_from_Scratch/en-US/Ch-Active-Active.txt

```diff
@@ -1,755 +1,364 @@
-    Masters: [ pcmk-1 ]
-    Slaves: [ pcmk-2 ]
-ClusterIP    (ocf::heartbeat:IPaddr):    Started pcmk-2WebFS (ocf::heartbeat:Filesystem): Started pcmk-1
------
-endif::[]
-
-== Reconfigure Pacemaker for Active/Active ==
-
-Almost everything is in place. Recent versions of DRBD are capable of
-operating in Primary/Primary mode and the filesystem we're using is
-cluster aware. All we need to do now is reconfigure the cluster to take
-advantage of this.
-
-ifdef::pcs[]
-This will involve a number of changes, so we'll want work with a
-local cib file.
-
-[source,C]
+GFS2 requires that DLM be running, so we also need to set up new colocation
+and ordering constraints for it:
 ----
-# pcs cluster cib active_cfg
+[root@pcmk-1 ~]# pcs constraint colocation add WebFS dlm-clone INFINITY
+[root@pcmk-1 ~]# pcs constraint order dlm-clone then WebFS
 ----
-endif::[]
 
-ifdef::crmsh[]
-This will involve a number of changes, so we'll again use interactive
-mode.
-
-[source,C]
------
-# crm
-# cib new active
------
-endif::[]
+== Clone the IP address ==
 
 There's no point making the services active on both locations if we can't
-reach them, so lets first clone the IP address. Cloned IPaddr2 resources
-use an iptables rule to ensure that each request only gets processed by one of
-the two clone instances. The additional meta options tell the cluster how
-many instances of the clone we want (one "request bucket" for each node)
-and that if all other nodes fail, then the remaining node should hold all
-of them. Otherwise the requests would be simply discarded.
+reach them both, so let's clone the IP address.
 
+The *IPaddr2* resource agent has built-in intelligence for when it is configured
+as a clone. It will utilize a multicast MAC address to have the local switch
+send the relevant packets to all nodes in the cluster, together with *iptables
+clusterip* rules on the nodes so that any given packet will be grabbed by
+exactly one node. This will give us a simple but effective form of
+load-balancing requests between our two nodes.
 
-ifdef::pcs[]
+Let's start a new config, and clone our IP:
 ----
-# pcs -f active_cfg resource clone ClusterIP \
-     globally-unique=true clone-max=2 clone-node-max=2
+[root@pcmk-1 ~]# pcs cluster cib loadbalance_cfg
+[root@pcmk-1 ~]# pcs -f loadbalance_cfg resource clone ClusterIP \
+     clone-max=2 clone-node-max=2 globally-unique=true
 ----
 
-Notice when the ClusterIP becomes a clone, the constraints
+* `clone-max=2` says packets will be split this many ways. This should
+equal the number of nodes that can host the IP.
+* `clone-node-max=2` says that one node can run up to 2 instances
+of the clone. This should also equal the number of nodes that can
+host the IP, so that if any node goes down, another node can take over
+the failed node's "request bucket". Otherwise, requests intended for
+the failed node would be discarded.
+* `globally-unique=true` tells the cluster that one clone isn't identical
```

Comment:
globally-unique is also the trigger that turns on the hashing of packets so that only one node processes each request (even though both have the IP active)